### PR TITLE
First invoice should be critical

### PR DESCRIPTION
**Pull request by soffokl**
Closes: https://github.com/mysteriumnetwork/node/issues/2034

### CODE REVIEW

File: session/pingpong/invoice_tracker.go [hunk 1]
```diff
@@ -297,6 +302,28 @@ func (it *InvoiceTracker) Start() error {
...
+	timeout := time.After(wait)
+
+	for {
+		select {
+		case <-time.After(10 * time.Millisecond):
+			it.invoiceLock.Lock()
+			if it.firstInvoicePaid {
+				return nil
+			}
+			it.invoiceLock.Unlock()
```
Comment (vkuznecovas) [inline session/pingpong/invoice_tracker.go:316]: if first invocie is paid this lock will not get released.

File: session/manager.go [hunk 2]
```diff
@@ -190,6 +190,11 @@ func (manager *Manager) Start(session *Session, consumerID identity.Identity, co
 		}
 	}()
 
+	log.Info().Msg("Waiting for a first invoice to be paid")
+	if err := engine.WaitFirstInvoice(3 * time.Second); err != nil {
```
Comment (anjmao) [inline session/manager.go:194]: I would increase first invoice to at least 10 seconds as some consumers can be slow.

Comment (soffokl) [reply session/manager.go:194]: Left it for 3 seconds for compatibility time only, to not slow down connection from old nodes too much. I'll add TODO for it.

File: session/pingpong/invoice_payer.go [hunk 3]
```diff
@@ -250,7 +249,6 @@ func (ip *InvoicePayer) issueExchangeMessage(invoice crypto.Invoice) error {
 
 	ip.deps.EventBus.Publish(event.AppTopicInvoicePaid, event.AppEventInvoicePaid{
 		ConsumerID: ip.deps.Identity,
-		SessionID:  ip.deps.SessionID,
```
Comment (vkuznecovas) [inline session/pingpong/invoice_payer.go:253]: Why remove this?

Comment (soffokl) [reply session/pingpong/invoice_payer.go]: Returned it back. Setting now SessionID after session established.

**Pull request merged**